### PR TITLE
Convert to pnpm to fix autocompletions

### LINKED ISSUE #285: Hovering comments on the playground throws an error (and sometimes an overlay)
https://play.predr.ag/hackernews , open your dev console, then hover any comment

### PR DESCRIPTION
**Pull request by u9g**
By converting to pnpm we can now patch our dependencies which is amazing since most of our meaningful code lies in our dependencies.

#### Started depending on a bunch of new packages.

Explanation: We already depended on them, our package manager just wasn't aware of that since npm probably has some tricks pnpm doesn't, or something along those lines.

#### Update `react-router-dom`

Explanation: It was causing some issues with pnpm, but I also don't think it causes any issues to update it in this pr.

#### Use `react-router-dom` in `experiments/browser_based_querying/src/QueryFragmentAdapter.tsx`

Explanation: We used `react-router-dom` and `react-router` together, which caused problems for pnpm, so we had to pick one according to SO, and it was easier to move to `react-router-dom`.

### Patches I've added:

`experiments/browser_based_querying/patches/graphql-language-service@5.1.7.patch` => The change to `esm/interface/getAutocompleteSuggestions.js` will fix autocompletions giving a directive when you wanted to autocomplete a field. Should still allow autocompleting directives after fields.

The change to `esm/interface/getDiagnostics.js` will close #285.

`experiments/browser_based_querying/patches/monaco-editor@0.20.0.patch` => Stop randomly logging from the lexer.

### CODE REVIEW

**Comment by u9g** [comment]
To note: we get these warnings on install, but they can be fixed in a future pr:
```
 WARN  deprecated core-js@3.22.5: core-js@<3.23.3 is no longer maintained and not recommended for usage due to the number of issues. Because of the V8 engine whims, feature detection in old core-js versions could cause a slowdown up to 100x even if nothing is polyfilled. Some versions have web compatibility issues. Please, upgrade your dependencies to the actual version of core-js.
Packages: -1
-
Progress: resolved 717, reused 717, downloaded 0, added 1, done
 WARN  Issues with peer dependencies found
.
└─┬ monaco-graphql 1.2.0
  └── ✕ unmet peer prettier@"^2.8.0 || ^3.0.0": found 2.6.2
```

File: experiments/browser_based_querying/README.md
Comment (u9g) [reply experiments/browser_based_querying/README.md:14]: It comes with node: https://nodejs.org/api/corepack.html#corepack

Comment (obi1kenobi) [reply experiments/browser_based_querying/README.md:14]: TIL! thanks

**Comment by obi1kenobi** [comment]
> We used react-router-dom and react-router together, which caused problems for pnpm

Can you say more about the problems here and why we needed to switch?

**Comment by u9g** [comment]
> > We used react-router-dom and react-router together, which caused problems for pnpm
> 
> Can you say more about the problems here and why we needed to switch?

I'd love to give more info, but to be honest I don't know exactly what the problem was, I just know that switching away from using both at once solved the problem described here: https://stackoverflow.com/a/73713976

File: experiments/browser_based_querying/README.md
Comment (obi1kenobi) [inline experiments/browser_based_querying/README.md:15]: Can you make this use a full command so it's more obvious what it does?

File: experiments/browser_based_querying/README.md
Comment (obi1kenobi) [inline experiments/browser_based_querying/README.md:11]: Do people need to install `pnpm` separately?

Comment (u9g) [reply experiments/browser_based_querying/README.md:11]: No, unless they installed node from homebrew: https://pnpm.io/installation#using-corepack

Comment (u9g) [reply experiments/browser_based_querying/README.md:11]: Saying the unless you installed node from homebrew part feels like an implementation detail that can change later

File: experiments/browser_based_querying/package.json
Comment (u9g) [reply experiments/browser_based_querying/package.json:38]: Yes, it's not used. I think it was conflicting with something so I removed it, but it doesn't matter really since it's not used. I can add it back then remove it in another pr if that would be better.

File: experiments/browser_based_querying/patches/graphql-language-service@5.1.7.patch
Comment (u9g) [reply experiments/browser_based_querying/patches/graphql-language-service@5.1.7.patch:27]: Yes it's machine generated.

File: experiments/browser_based_querying/patches/graphql-language-service@5.1.7.patch
Comment (u9g) [reply experiments/browser_based_querying/patches/graphql-language-service@5.1.7.patch:25]: You're right, I had to change the invariant in the pr: https://github.com/graphql/graphiql/pull/3397

This isn't a huge problem for us, we can just get rid of this patch and wait until that pr is merged.

File: experiments/browser_based_querying/README.md
Comment (obi1kenobi) [reply experiments/browser_based_querying/README.md:11]: It would be good to add a note for homebrew. We don't wanna have to debug broken workflows later if we already know a possible issue that might come up.

File: experiments/browser_based_querying/patches/graphql-language-service@5.1.7.patch
Comment (u9g) [reply experiments/browser_based_querying/patches/graphql-language-service@5.1.7.patch]: (it does work in it's current form too though...)

File: experiments/browser_based_querying/patches/monaco-editor@0.20.0.patch
Comment (u9g) [reply experiments/browser_based_querying/patches/monaco-editor@0.20.0.patch]: Added a comment to clarify what's actually changed and commented it out line by line so it's obvious exactly what's commented out.

**Review by obi1kenobi — APPROVED** [review]

**Pull request merged**
